Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.cpp

```diff
@@ -430,6 +431,14 @@ void RemoteImageBufferProxy::prepareForBackingStoreChange()
         backend->ensureNativeImagesHaveCopiedBackingStore();
 }
 
+void RemoteImageBufferProxy::remoteBufferWasDestroyed()
+{
+    if (m_remoteBufferManager == RemoteBufferManager::Self && m_remoteRenderingBackendProxy)
```

Comment:
I think the changes related to this are somehow out of place and hard to maintain.
I don't  immediately understand the  logic "if proxy is part of set, don't recreate, otherwise recreate".
I think the recreation logic should be understandable without understanding the sets.

To me the m_remoteBufferManager additions seem overly complicated and should be perhaps removed altogether. Definitively the name "Manager" to indicate "Yes, this is part of some set" is odd?